presto-main/src/main/java/com/facebook/presto/execution/SharedBuffer.java

```diff
@@ -182,9 +172,15 @@ public SharedBufferInfo getInfo()
 
         ImmutableList.Builder<BufferInfo> infos = ImmutableList.builder();
         for (NamedBuffer namedBuffer : namedBuffers.values()) {
-            infos.add(namedBuffer.getInfo());
+            BufferInfo info = namedBuffer.getInfo();
+            infos.add(info);
         }
-        return new SharedBufferInfo(state.get(), masterSequenceId.get(), pagesAdded.get(), infos.build());
+
+        long totalBufferedBytes = partitionToPageBuffer.values().stream().mapToLong(PageBuffer::getBufferedBytes).sum();
+        long totalBufferedPages = partitionToPageBuffer.values().stream().mapToLong(PageBuffer::getBufferedPageCount).sum();
+        long totalQueuedPages = partitionToPageBuffer.values().stream().mapToLong(PageBuffer::getQueuedPageCount).sum();
+        long totalPagesSent = partitionToPageBuffer.values().stream().mapToLong(PageBuffer::getPageCount).sum();
+        return new SharedBufferInfo(state.get(), state.get().canAddBuffers(), state.get().canAddPages(), totalBufferedBytes, totalBufferedPages, totalQueuedPages, totalPagesSent, infos.build());
```

Comment:
you should call `state.get()` once so you have consistent results (this code is lock free)